### PR TITLE
Handle Single Cell and Spatial Metadata Parsing, and Sample-Library Combining

### PR DESCRIPTION
**Pull request by avrohomgottlieb**
## Issue Number

#645 (main one)
#635 (secondary one)

## Purpose/Implementation Notes

This PR is the first step in the refactoring of the metadata handling logic in the `Project::load_data` method. For the most part the logic itself was untouched but rather moved around and abstracted into generic sub-methods with a single `handle_samples_metadata` now being callable from within `load_data` itself. The next steps will be to tuck multiplex logic into `handle_samples_metadata`, while removing all of the file writing functionality from `handle_samples_metadata` to its own method.

## Types of changes

What types of changes does your code introduce?

- Refactor (non-breaking change which does not add functionality but improves code organization and design)

## Functional tests

No tests added or modified (inputs and outputs are the same, just the implementation was refactored, so tests remained the same).

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshots

N/A

### CODE REVIEW

File: api/scpca_portal/models/project.py [hunk 1]
```diff
@@ -1155,6 +1101,171 @@ def update_spatial_data(future):
...
+                sample_seq_units.add(library_json["seq_unit"].strip())
+                sample_technologies.add(library_json["technology"].strip())
+
+                libraries_metadata[Sample.Modalities.SINGLE_CELL].append(
+                    library_json
+                ) if "spatial" not in str(filename_path) else libraries_metadata[
+                    Sample.Modalities.SPATIAL
+                ].append(
+                    library_json
+                )
```
Comment (davidsmejia) [inline api/scpca_portal/models/project.py:1185]: Could we rewrite this for clarity?

File: api/scpca_portal/models/project.py [hunk 2]
```diff
@@ -1155,6 +1101,171 @@ def update_spatial_data(future):
...
+            updated_sample_metadata["technologies"] = ", ".join(
+                sorted(sample_technologies, key=str.lower)
+            )
+
+        return (updated_samples_metadata, libraries_metadata)
+
+    def combine_metadata(self, updated_samples_metadata, libraries_metadata, sample_id):
+        combined_metadata = {Sample.Modalities.SINGLE_CELL: [], Sample.Modalities.SPATIAL: []}
+
+        for modality in [Sample.Modalities.SINGLE_CELL, Sample.Modalities.SPATIAL]:
```
Comment (davidsmejia) [inline api/scpca_portal/models/project.py:1200]:
```suggestion
        for modality in combined_metadata.keys():
```

**Review by davidsmejia — APPROVED** [review]
This looks good, made a couple suggestions.

File: api/scpca_portal/models/project.py (hunk 1 above)
Comment (avrohomgottlieb) [reply api/scpca_portal/models/project.py]: Good call. I think I might have been auto-formatted here. In any case I think a normal `if else` makes more sense here then the python ternary.

**Pull request merged**